packages/boxel-ui/addon/src/components/multi-select/selected-item.gts

```diff
@@ -0,0 +1,68 @@
+import { IconX } from '@cardstack/boxel-ui/icons';
+import { fn } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import type { Select } from 'ember-power-select/components/power-select';
+
+// Inside the trigger component, the selected item component is called within in ember-power-select
+// It only passes option, select and extra as arguments
+// We follow the same convention when defining our own selected item component
+export interface SelectedItemSignature<ItemT> {
+  Args: {
+    extra?: any;
+    option: any;
+    select: Select & {
+      actions: {
+        remove: (item: ItemT) => void;
+      };
+    };
+  };
+  Blocks: {
+    default: [any];
+  };
+  Element: HTMLDivElement;
+}
+
+export default class BoxelSelectedItem<ItemT> extends Component<
+  SelectedItemSignature<ItemT>
+> {
+  @action
+  remove(item: ItemT, event: MouseEvent) {
+    // Do not remove these event
```

Comment:
Noted: The point here is to prevent propagation into the trigger that will cause the dropdown to close. This is under the circumstance that you remove the selected item by clicking its' x. I have added the comment.